src/topology/homotopy/induced_maps.lean

```diff
@@ -0,0 +1,184 @@
+@category_theory.functor.map _ _ _ _ (prod_to_prod_Top (Top.of $ ulift I) X)
+  (a₁, b₁) (a₂, b₂) (p₁, p₂)
+
+/-- The diagonal path `d` of a homotopy `H` on a path `p` -/
+def diagonal_path : from_top (H (0, x₀)) ⟶ H (1, x₁) :=
+(πₘ H.ulift_map).map (prod_to_prod_Top_I uhpath01 p)
+
+/-- The diagonal path, but starting from `f x₀` and going to `g x₁` -/
+def diagonal_path' : from_top (f x₀) ⟶ g x₁ :=
+hcast (H.apply_zero x₀).symm ≫ (H.diagonal_path p) ≫ hcast (H.apply_one x₁)
+
+/-- Proof that `f(p) = H(0 ⟶ 0, p)`, with the appropriate casts -/
+lemma apply_zero_path : (πₘ f).map p = hcast (H.apply_zero x₀).symm
+  ≫ (πₘ H.ulift_map).map (prod_to_prod_Top_I (𝟙 (ulift.up 0)) p)
+  ≫ hcast (H.apply_zero x₁) :=
+begin
+  apply quotient.induction_on p,
+  intro p',
+  apply @eq_path_of_eq_image _ _ _ _ H.ulift_map _ _ _ _ _ ((path.refl (ulift.up _)).prod p'),
+  simp,
+end
+
+/-- Proof that `g(p) = H(1 ⟶ 1, p)`, with the appropriate casts -/
+lemma apply_one_path : (πₘ g).map p = hcast (H.apply_one x₀).symm
+  ≫ ((πₘ H.ulift_map).map (prod_to_prod_Top_I (𝟙 (ulift.up 1)) p))
+  ≫ hcast (H.apply_one x₁) :=
+begin
+  apply quotient.induction_on p,
+  intro p',
+  apply @eq_path_of_eq_image _ _ _ _ H.ulift_map _ _ _ _ _ ((path.refl (ulift.up _)).prod p'),
+  simp,
+end
+
+/-- Proof that `H.to_path x = H(0 ⟶ 1, x ⟶ x)`, with the appropriate casts -/
+lemma to_path_eq (x : X) : ⟦H.to_path x⟧ =
+  hcast (H.apply_zero x).symm ≫
+  (πₘ H.ulift_map).map (prod_to_prod_Top_I uhpath01 (𝟙 x)) ≫
+  hcast (H.apply_one x) :=
+begin
+  dunfold prod_to_prod_Top_I uhpath01,
+  simp only [id_eq_path_refl, prod_to_prod_Top_map, path.homotopic.prod_lift, map_eq,
+    ← path.homotopic.map_lift, path_cast_right, path_cast_left],
+  refl,
+end
+
+/- Finally, we show `d = f(p) ≫ H₁ = H₀ ≫ g(p)` -/
+lemma eq_diag_path :
+  (πₘ f).map p ≫ ⟦H.to_path x₁⟧ = H.diagonal_path' p ∧
+  (⟦H.to_path x₀⟧ ≫ (πₘ g).map p : from_top (f x₀) ⟶ g x₁) = H.diagonal_path' p :=
+begin
+  rw [H.apply_zero_path, H.apply_one_path, H.to_path_eq, H.to_path_eq],
+  dunfold prod_to_prod_Top_I,
+  split; { slice_lhs 2 5 { simp [← category_theory.functor.map_comp], }, refl, },
+end
+
+end continuous_map.homotopy
+
+namespace fundamental_groupoid_functor
+open category_theory
+open_locale fundamental_groupoid
+local attribute [instance] path.homotopic.setoid
+
+variables {X Y : Top.{u}} {f g : C(X, Y)} (H : continuous_map.homotopy f g)
+
+/-- Given a homotopy H : f ∼ g, we have an associated natural isomorphism between the induced
+functors `f` and `g` -/
+def homotopic_maps_nat_iso : πₘ f ⟶ πₘ g :=
+{ app := λ x, ⟦H.to_path x⟧,
+  naturality' := λ x y p, by rw [(H.eq_diag_path p).1, (H.eq_diag_path p).2] }
+
+instance : is_iso (homotopic_maps_nat_iso H) := by apply nat_iso.is_iso_of_is_iso_app
+
+open_locale continuous_map
+
+/-- Homotopy equivalent topological spaces have equivalent fundamental groupoids. -/
+def equiv_of_homotopy_equiv (hequiv : X ≃ₕ Y) : (πₓ X).α ≌ (πₓ Y).α :=
```

Comment:
It's been a long time since I thought about this, but my memory is that somewhen we decided to avoid using `.α` as much as possible in favour of the coercion to type.

Could you perhaps try this out?